### PR TITLE
[feat]: [PL-25236]: Script to copy entities from one project to another

### PR DESCRIPTION
**Pull request by rathodmeetsatish**
Script to copy entities from one project to another

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>
  
- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>
  
  You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`
  
- Compile: `trigger compile`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
</details>

---
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/33723)

### CODE REVIEW

File: scripts/copy-entities/copy_entities_script.py [hunk 1]
```diff
@@ -0,0 +1,264 @@
...
+# that can be found in the licenses directory at the root of this repository, also available at
+# https://polyformproject.org/wp-content/uploads/2020/05/PolyForm-Free-Trial-1.0.0.txt.
+
+import http.client
+import mimetypes
+import json
+import yaml
+import sys
+from codecs import encode
+from abc import *
```
Comment (abhinav-harness) [inline scripts/copy-entities/copy_entities_script.py:12]: remove

File: scripts/copy-entities/copy_entities_script.py [hunk 2]
```diff
@@ -0,0 +1,264 @@
...
+source_query_param = routing_and_accountId_param+"&orgIdentifier="+from_orgIdentifier+"&projectIdentifier="+from_projectIdentifier
+conn = http.client.HTTPSConnection("app.harness.io")
+headers = {
+  'x-api-key': apikey,
+  'content-type': 'application/json'
+  }
+global success_count
+global failure_count
+
+class ImportExport(ABC):
```
Comment (abhinav-harness) [inline scripts/copy-entities/copy_entities_script.py:37]: ABC is wrong name

**Review by rathodmeetsatish — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/33723)* status: 1 file(s) to review. 1. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`scripts/copy-entities/copy_entities_script.py` line 37 at r1](https://reviewable.io/reviews/harness/harness-core/33723#-N3OdkZ-MIQbQ6-uQ93N-r1-37:-N3YftGw4QbG2txI2loq:b-ek0scc) ([raw file](https://github.com/harness/harness-core/blob/aef1f00b644333933c132a286882961a41bf8072/scripts/copy-entities/copy_entities_script.py#L37)):*
<details><summary><i>Previously, abhinav-harness (Abhinav Singh) wrote…</i></summary><blockquote>

ABC is wrong name
</blockquote></details>

its the library name

**Review by abhinav-harness — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/33723)* status: 1 file(s) to review. 1 change requests. 0 of 2 approvals obtained. 

___
*[`scripts/copy-entities/copy_entities_script.py` line 261 at r1](https://reviewable.io/reviews/harness/harness-core/33723#-N3Ypwv9EgwjoDB_NFQ6:-N3Ypwv9EgwjoDB_NFQ7:bu95qq0) ([raw file](https://github.com/harness/harness-core/blob/aef1f00b644333933c132a286882961a41bf8072/scripts/copy-entities/copy_entities_script.py#L261)):*
> ```Python
> print("Successfully copied "+str(success_failure_count_connector[0])+" connectors and got error while copying "+str(success_failure_count_connector[1])+" connectors")
> print("Successfully copied "+str(success_failure_count_template[0])+" templates and got error while copying "+str(success_failure_count_template[1])+" templates")
> print("Successfully copied "+str(success_failure_count_pipeline[0])+" pipelines and got error while copying "+str(success_failure_count_pipeline[1])+" pipelines")
> ```

Clean the code here - 
```
make a list add there and use for loop to publish and for loop to call main_exoprt for those entity
```



_Code quote:_
```Python
success_failure_count_secret = main_export("Secret")

success_failure_count_service= main_export("Service")

success_failure_count_environment = main_export("Environment")

success_failure_count_connector = main_export("Connector")

success_failure_count_template = main_export("Template")

success_failure_count_pipeline = main_export("Pipeline")

print("Successfully copied "+str(success_failure_count_secret[0])+" secret and got error while copying "+str(success_failure_count_secret[1])+" secrets")
print("Successfully copied "+str(success_failure_count_service[0])+" services and got error while copying "+str(success_failure_count_service[1])+" services")
print("Successfully copied "+str(success_failure_count_environment[0])+" environments and got error while copying "+str(success_failure_count_connector[1])+" environments")
print("Successfully copied "+str(success_failure_count_connector[0])+" connectors and got error while copying "+str(success_failure_count_connector[1])+" connectors")
print("Successfully copied "+str(success_failure_count_template[0])+" templates and got error while copying "+str(success_failure_count_template[1])+" templates")
print("Successfully copied "+str(success_failure_count_pipeline[0])+" pipelines and got error while copying "+str(success_failure_count_pipeline[1])+" pipelines")
```

**Review by abhinav-harness — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/33723)* status: 1 file(s) to review. 1 change requests. 0 of 2 approvals obtained. 

___
*[`scripts/copy-entities/copy_entities_script.py` line 93 at r1](https://reviewable.io/reviews/harness/harness-core/33723#-N3YqNFF2pKpSq701pYY:-N3YqNFF2pKpSq701pYZ:b-prfgjw) ([raw file](https://github.com/harness/harness-core/blob/aef1f00b644333933c132a286882961a41bf8072/scripts/copy-entities/copy_entities_script.py#L93)):*
> ```Python
>       url_create_input_set = input_set_endpoint + "?routingId="+accountIdentifier+"&accountIdentifier="+accountIdentifier+"&orgIdentifier="+to_orgIdentifier+"&pipelineIdentifier="+pipelineIdentifier+"&projectIdentifier=" + to_projectIdentifier
>       create_and_print_entity(url_create_input_set, new_payload_json)
> ```

move this below implementors



_Code quote:_
```Python
def get_response_data(request_type, url, payload):
  conn.request(request_type, url, payload, headers)
  res = conn.getresponse()
  data = res.read()
  return json.loads(data.decode("utf-8"))

def modify_payload(payload, entity_type):
  payload[entity_type]["projectIdentifier"] = to_projectIdentifier
  payload[entity_type]["orgIdentifier"] = to_orgIdentifier
  return json.dumps(payload)

def list_entity(url, filterType):
  payload = json.dumps({"filterType":filterType})
  return get_response_data("POST", url, payload)

def create_and_print_entity(url, payload):
  response_create_dict = get_response_data("POST", url, payload)
  print(response_create_dict)
  print("\n")
  return response_create_dict

def export_input_set(pipelineIdentifier):
  url = input_set_endpoint+"?routingId="+accountIdentifier+"&accountIdentifier="+accountIdentifier+"&orgIdentifier="+from_orgIdentifier+"&projectIdentifier="+from_projectIdentifier+"&pipelineIdentifier="+pipelineIdentifier+"&pageIndex=0&pageSize=20"
  response_dict = get_response_data("GET", url, "")

  for j in range(0,response_dict["data"]["totalPages"]):
    url_paginated = input_set_endpoint+"?routingId="+accountIdentifier+"&accountIdentifier="+accountIdentifier+"&orgIdentifier="+from_orgIdentifier+"&projectIdentifier="+from_projectIdentifier+"&pipelineIdentifier="+pipelineIdentifier+"&pageIndex="+str(j)+"&pageSize=20"
    response_dict = get_response_data("GET", url_paginated, "")

    for i in range(0,response_dict["data"]["pageItemCount"]):
      url_get_input_set = input_set_endpoint +"/"+ response_dict["data"]["content"][i]["identifier"] + "?routingId="+accountIdentifier+"&accountIdentifier="+accountIdentifier+"&orgIdentifier="+from_orgIdentifier+"&pipelineIdentifier="+pipelineIdentifier+"&projectIdentifier="+from_projectIdentifier
      response_get_input_set = get_response_data("GET", url_get_input_set, "")
      new_payload_json = modify_payload(yaml.safe_load(response_get_input_set["data"]["inputSetYaml"]), "inputSet")
      url_create_input_set = input_set_endpoint + "?routingId="+accountIdentifier+"&accountIdentifier="+accountIdentifier+"&orgIdentifier="+to_orgIdentifier+"&pipelineIdentifier="+pipelineIdentifier+"&projectIdentifier=" + to_projectIdentifier
      create_and_print_entity(url_create_input_set, new_payload_json)
```

**Review by rathodmeetsatish — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/33723)* status: 1 file(s) to review. 1 change requests. 2. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`scripts/copy-entities/copy_entities_script.py` line 93 at r1](https://reviewable.io/reviews/harness/harness-core/33723#-N3YqNFF2pKpSq701pYY:-N3sdSQXB9TJVgAnvczw:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/aef1f00b644333933c132a286882961a41bf8072/scripts/copy-entities/copy_entities_script.py#L93)):*
<details><summary><i>Previously, abhinav-harness (Abhinav Singh) wrote…</i></summary><blockquote>

move this below implementors
</blockquote></details>

Done.
___
*[`scripts/copy-entities/copy_entities_script.py` line 261 at r1](https://reviewable.io/reviews/harness/harness-core/33723#-N3Ypwv9EgwjoDB_NFQ6:-N3sdTSD1-X18Rnkay67:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/aef1f00b644333933c132a286882961a41bf8072/scripts/copy-entities/copy_entities_script.py#L261)):*
<details><summary><i>Previously, abhinav-harness (Abhinav Singh) wrote…</i></summary><blockquote>

Clean the code here - 
```
make a list add there and use for loop to publish and for loop to call main_exoprt for those entity
```
</blockquote></details>

Done.

**Review by abhinav-harness — DISMISSED** [review]
> Reviewed 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/33723)* status: 1 file(s) to review. 1 of 2 approvals obtained.

**Review by abhinav-harness — APPROVED** [review]
> Reviewed 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/33723)* status: 1 file(s) to review. 1 of 2 approvals obtained.

**Review by mankrit-singh — APPROVED** [review]
> Reviewed 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/33723)* status: :shipit: complete! ✓

**Pull request merged**